src/util/helpers.ts

```diff
@@ -75,12 +75,20 @@ export const toBitcoinAmount = (amountInSatoshi: string): string => {
   let value;
 
   if (bnAmountInSatoshi.lt(bnSatoshis)) {
-    value = bnAmountInSatoshi.div(bnSatoshis).toNumber(); // Okay to do toNumber as we're dealing with TKL and not satoshis
-  } else {
     value = parseInt(amountInSatoshi, 10) / SATOSHIS;
+    value = Number(value.toFixed(Config.DECIMAL)).toString();
+  } else {
+    value = bnAmountInSatoshi.div(bnSatoshis).toNumber(); // Okay to do toNumber as we're dealing with TKL and not satoshis
+
+    value = parseFloat(
+      `${value.toString()}.${bnAmountInSatoshi
+        .toNumber()
+        .toString()
+        .substring(value.toString().length)}`
+    );
```

Comment:
the problem with this line is that if the amount in satoshi is too big the `.toNumber()` will fail. 

you need to subtract the rounded `value` from the big number until only less than 1 TKL (100 million satoshi) is left, aka a module.

Here's an example with 1.67 TKL. The result of the equation which 

```js
const mod = new BN("167000000").mod(new BN("100000000"));

You can then do toNumber safely as we know this is less than one million satoshis;
mod.toNumber();

// You can even use toBitcoin here and it will yield 0.67
toBitcoin(mod.toNumber()); 

// You can then glue the two things together and you get a safe big-number proof amount in TKL
const total = (bnAmountInSatoshi.div(bnSatoshis).toNumber() + toBitcoin(module.toNumber()).toFixed(Config.DECIMAL);
```

Reply:
`.substring(value.toString().length)` this part of the line will "subtract" the amount before the dot. So let's say the transaction is for 1.67 TKL.

The 1 before the dot is already fetched correctly in the line before, so the `${value.toString()}` is always going to pick the correct amount, now we need to pick the rest of the number after the dot.

Since we don't know how many numbers are before the dot, I decided to convert the number before the dot to string and use `substring` method with the length of the value to properly subtract the first numbers.

- For a transaction of 1.67, it would remove the first number;
- For a transaction of 11.67, it would remove the first two numbers;
- For a transaction of 111.67, the three first numbers, and so on...

Does it make sense? Or do you think we would still need to subtract the number the way you suggested?